src/file_resolver.rs

```diff
@@ -199,49 +190,69 @@ impl FileResolver {
                 let path = base.join(&path);
 
                 if let Some(file) = self.try_file(filename, &path, *import_no)? {
+                    // No ambiguity possible, so just return
                     return Ok(file);
                 }
             }
 
             return Err(format!("file not found '{}'", filename.to_string_lossy()));
         }
 
-        if let Some(file) = self.try_file(filename, &path, None)? {
-            return Ok(file);
-        } else if path.is_absolute() {
-            return Err(format!("file not found '{}'", filename.to_string_lossy()));
+        if parent.is_none() {
+            if let Some(file) = self.try_file(filename, &path, None)? {
+                return Ok(file);
+            } else if path.is_absolute() {
+                return Err(format!("file not found '{}'", filename.to_string_lossy()));
+            }
         }
 
         // first check maps
-        let mut iter = path.iter();
-        if let Some(first_part) = iter.next() {
-            let relpath: &PathBuf = &iter.collect();
-
-            for import_no in 0..self.import_paths.len() {
-                if let (Some(mapping), import_path) = &self.import_paths[import_no] {
-                    if first_part == mapping {
-                        let path = import_path.join(relpath);
-
-                        if let Some(file) = self.try_file(filename, &path, Some(import_no))? {
-                            return Ok(file);
-                        }
-                    }
+
+        let mut remapped = path.clone();
```

Comment:
We could just make `path` mutable, can clean this up if it's desirable.

Basically, what we are doing here is iterating over remappings and applying them if possible. This leaves us with the final applicable remapping being applied, which matches Solc's semantics